docker/Dockerfile

```diff
@@ -70,48 +80,26 @@ COPY ./requirements-tests.txt .
 RUN pip install -r requirements-tests.txt
 
 # Download Vegeta tool for benchmarking
-ENV VEGETA_DOWNLOAD_SHA256 2f0a69d0ae6f0bf268b7f655bd37c0104d5568d5b2bc45bbb2c405266f74e33d
-ENV VEGETA_DOWNLOAD_URL https://github.com/tsenart/vegeta/releases/download/v6.1.1/vegeta-v6.1.1-linux-amd64.tar.gz
-
 RUN curl -fsSL "$VEGETA_DOWNLOAD_URL" -o vegeta.tar.gz \
     && echo "$VEGETA_DOWNLOAD_SHA256  vegeta.tar.gz" | sha256sum -c - \
         && tar -C /usr/local/bin -xzf vegeta.tar.gz \
         && rm vegeta.tar.gz
 
-# Compile AR, prepare it for running
-# This mimics the settings from dcos-image except for '--with-debug'
+# Prepare Openresty. Compilation is done in Makefile itself so that
+# this container can be reused during DC/OS build.
 RUN set -ex \
     && curl -fsSL "$OPENRESTY_DOWNLOAD_URL" -o openresty.tar.gz \
     && echo "$OPENRESTY_DOWNLOAD_SHASUM  openresty.tar.gz" | shasum -c - \
     && mkdir -pv $OPENRESTY_DIR \
     && tar --strip-components=1 -C $OPENRESTY_DIR -xzf openresty.tar.gz \
-    && rm openresty.tar.gz \
-    && cd $OPENRESTY_DIR \
-    && ./configure \
-        "--prefix=$AR_BIN_DIR" \
-        --with-ipv6 \
-        --with-debug \
-        --with-file-aio \
-        --with-http_gunzip_module \
-        --with-http_gzip_static_module \
-        --without-mail_pop3_module \
-        --without-mail_imap_module \
-        --without-mail_smtp_module \
-        --with-http_ssl_module \
-        --with-luajit \
-    && make \
-    && make install \
-    && mv -v ${AR_BIN_DIR}/nginx/conf/ ${AR_BIN_DIR}/nginx/conf.orig/ \
-    && ln -s ${AR_CONF_DIR} ${AR_BIN_DIR}/nginx/conf
+    && rm openresty.tar.gz
+
+COPY build-resty.sh $OPENRESTY_COMPILE_SCRIPT
 
 # Some AR variables that mimic DC/OS:
 COPY ./iam.jwt-key.pub /usr/local/
```

Comment:
If you want to save same additional layers we could replace COPY with ADD directory instruction and running a bash script that would distribute files to right location.